### PR TITLE
Updating the hot reload page

### PR DESCRIPTION
**Pull request by sfshaza2**
Staged:  https://sz-flutter.firebaseapp.com/docs/development/tools/hot-reload

Fixes https://github.com/flutter/website/issues/3625
Fixes https://github.com/flutter/website/issues/3566. (maybe)
Fixes https://github.com/flutter/website/issues/2920
Fixes https://github.com/flutter/website/issues/2056
Fixes https://github.com/flutter/website/issues/1619

I'll be emailing some potential reviewers offline.

### CODE REVIEW

File: src/docs/development/tools/hot-reload.md [hunk 1]
```diff
@@ -37,18 +35,30 @@ you'll see a message in the console similar to:
...
-after the change. Specifically, a hot reload causes all the existing widgets to
-rebuild. Only code involved in the rebuilding of the widgets are automatically
-re-executed.
+{{site.alert.secondary}}
+  **Hot reload vs hot restart** In the simplest terms,
+  a hot reload refreshes your running app without losing state.
+  A hot restart resets the app's state to its default values,
+  but is much faster than a cold start.
+
+  Flutter web supports hot restart but not
```
Comment (devoncarew) [inline src/docs/development/tools/hot-reload.md:50]: 'currently supports'?

File: src/docs/development/tools/hot-reload.md [hunk 2]
```diff
@@ -310,27 +293,50 @@ supported at all. These include:
   }
   ```
 
-In these situations,
-hot reload generates a diagnostic message and fails without
-committing any changes.
+* Prior to Flutter SDK 1.17, if you modified a
```
Comment (devoncarew) [inline src/docs/development/tools/hot-reload.md:296]: Do we need to document older behavior? Perhaps remove the mention of the pre-1.17 reload behavior wrt class hierarchy changes.

**Comment by InMatrix** [comment]
It would be great to cover the list of limitations in https://github.com/flutter/website/issues/2920 as much as possible in this revision. The list includes the following hot reload limitations:

- [ ] Changing native code 
- [ ] Changing code related to routing (flutter/flutter/issues/43574)
- [ ] Changing packages
- [ ] Reloading fonts (flutter/flutter#49230)
- [ ] Additional limitations when using statement management packages (#3566)
- [X] Change from StatelessWidget to StatefulWidget (flutter/flutter/issues/9811)

Among other refinements, the current PR covers the last one but not others. Except the issue about state management and hot reload, the rest seems to be specific enough to be mentioned in this doc.

File: src/docs/development/tools/hot-reload.md (hunk 1 above)
Comment (sfshaza2) [reply src/docs/development/tools/hot-reload.md:50]: Done.

**Comment by sfshaza2** [comment]
Staged:  https://sz-flutter.firebaseapp.com/docs/development/tools/hot-reload

**Comment by sfshaza2** [comment]
Hello reviewers! I'm interpreting your silence as tacit approval on the new page. I'm landing this and, if you later see a problem, please file an issue!  Thanks.

**Comment by InMatrix** [comment]
Sorry about the late response. Thanks for addressing most of the special cases we've discovered so far in the updated doc.

**Pull request merged**